packages/tracking/src/Tracking.php

```diff
@@ -66,20 +89,26 @@ function record_user_event( $event_type, $data = array(), $user = null ) {
 	/**
 	 * Record an event in Tracks - this is the preferred way to record events from PHP.
 	 *
-	 * @param mixed  $identity username, user_id, or WP_user object
-	 * @param string $event_name The name of the event
-	 * @param array  $properties Custom properties to send with the event
-	 * @param int    $event_timestamp_millis The time in millis since 1970-01-01 00:00:00 when the event occurred
+	 * @param mixed  $user                   username, user_id, or WP_user object.
+	 * @param string $event_name             The name of the event.
+	 * @param array  $properties             Custom properties to send with the event.
+	 * @param int    $event_timestamp_millis The time in millis since 1970-01-01 00:00:00 when the event occurred.
 	 *
 	 * @return bool true for success | \WP_Error if the event pixel could not be fired
 	 */
-	function tracks_record_event( $user, $event_name, $properties = array(), $event_timestamp_millis = false ) {
+	public function tracks_record_event( $user, $event_name, $properties = array(), $event_timestamp_millis = false ) {
 
 		// We don't want to track user events during unit tests/CI runs.
 		if ( $user instanceof \WP_User && 'wptests_capabilities' === $user->cap_key ) {
 			return false;
 		}
 
+		// Don't track users who have opted our, or not agreed to the TOS, or sites in development mode.
+		$status = new Status();
+		if ( ! \Jetpack::jetpack_tos_agreed() || ! empty( $_COOKIE['tk_opt-out'] ) || $status->is_development_mode() ) {
```

Comment:
I don't think we need to be that detailed:
- Checking for the cookie isn't necessary as the opt-out is covered on WordPress.com. See p9SXHV-6Y-p2 to find out more.
- `jetpack_tos_agreed` checks if Jetpack is active, so I think that would cover development mode in most cases, except for folks that had Jetpack active, and thus gave their consent for tracking, and then activated development mode. I think that's fine though, as they gave their consent.

```suggestion
		if ( ! \Jetpack::jetpack_tos_agreed() ) {
```

Reply:
Removing the cookie check make sense. I'll push that up while I'm in there.